angular2-modal breaks with  RC4 release

* **I'm submitting a ...**
[ Y] bug report
[ ] feature request
[ ] question about the decisions made in the repository

* **Do you want to request a *feature* or report a *bug*?**
Bug


* **What is the current behavior?**
Post Angular RC4 update, Angular2-modal version `1.0.7` throws compilation  error when compiled with TypeScript compiler. 

```
.../node_modules/angular2-modal/platform-browser/outside-event-plugin.d.ts(1,10): 

error TS2305: Module '"...node_modules/@angular/platform-browser/index"' has no exported member 'DomEventsPlugin'.
```
This may be due to this [RC4 change](https://github.com/angular/angular/blob/master/CHANGELOG.md)

> Previously deprecated BROWSER_PROVIDERS was completely removed from platform-browser.

--- a/README.md
+++ b/README.md
@@ -1,6 +1,6 @@
 # UPDATE: Angular 2 Modal version 1 is out.
 
-# Angular 2 (rc.2): AIO Modal / Dialog window 
+# Angular 2 (rc.4): AIO Modal / Dialog window
 
 A fully generic, customizable and fluent modal window implementation for Angular.
 `angular2-modal` is UI platform/framework agnostic, **plugins**** are used to describe a UI implementation (e.g: Bootstrap)  
@@ -16,15 +16,15 @@ Take 5 minutes to read a [quick walk through](https://github.com/shlomiassaf/ang
   * [Bootstrap]()
   * [Vex](http://github.hubspot.com/vex/docs/welcome/)
   * POC implementation of JS Native modal (window.alert/prompt/confirm) to demonstrate a hostile takeover :)
-  
-  
-## Built with angular 2 RC.2
+
+
+## Built with angular 2 RC.4
 
 ## Features  
 
   - Easy to use API via Fluent API Presets (alert, propmt, confirm)
   - Extendable via plugins.  
-  
+
 #### Bootstrap / VEX
   - Customizable with components, Presets and more...  
   - Select cancel/quit key.
@@ -107,7 +107,7 @@ Test against `universal-starter` to enable universal support.
 Support `OnPush` with an observable api model (fluent observables)
 
 # Known bugs
-### The dialog closes when removing the target DOM element in a click event 
+### The dialog closes when removing the target DOM element in a click event
 ref [issue#111](https://github.com/shlomiassaf/angular2-modal/issues/111)
 
 To avoild this problem use `event.stopPropagation();` or put the element removal inside a `setTimeout` call

--- a/package.json
+++ b/package.json
@@ -31,16 +31,16 @@
     "start": "npm run server:dev"
   },
   "peerDependencies": {
-    "@angular/core": "^2.0.0-rc.2",
-    "@angular/compiler": "^2.0.0-rc.2",
-    "@angular/common": "^2.0.0-rc.2",
-    "@angular/platform-browser": "^2.0.0-rc.2",
-    "@angular/platform-browser-dynamic": "^2.0.0-rc.2",
+    "@angular/core": "^2.0.0-rc.4",
+    "@angular/compiler": "^2.0.0-rc.4",
+    "@angular/common": "^2.0.0-rc.4",
+    "@angular/platform-browser": "^2.0.0-rc.4",
+    "@angular/platform-browser-dynamic": "^2.0.0-rc.4",
     "rxjs": "5.0.0-beta.6",
     "zone.js": "^0.6.12"
   },
   "devDependencies": {
-    "@angular/router": "^3.0.0-alpha.7",
+    "@angular/router": "^3.0.0-beta.2",
     "angular-cli": "0.0.31",
     "async": "^1.5.2",
     "bootstrap": "^3.3.6",

--- a/src/bootstrap.ts
+++ b/src/bootstrap.ts
@@ -1,7 +1,7 @@
 import { ComponentRef, PLATFORM_DIRECTIVES } from '@angular/core';
 import { bootstrap } from '@angular/platform-browser-dynamic';
 import { LocationStrategy, HashLocationStrategy } from '@angular/common';
-import { ELEMENT_PROBE_PROVIDERS } from '@angular/platform-browser';
+import { BROWSER_APP_PROVIDERS } from '@angular/platform-browser';
 import { ROUTER_DIRECTIVES } from '@angular/router';
 
 import { MODAL_BROWSER_PROVIDERS } from './components/angular2-modal/platform-browser';
@@ -23,7 +23,7 @@ export function main(): Promise<ComponentRef<App>> {
       {provide: PLATFORM_DIRECTIVES, multi: true, useValue: ROUTER_DIRECTIVES},
       APP_ROUTER_PROVIDERS,
       {provide: LocationStrategy, useClass: HashLocationStrategy},
-      ELEMENT_PROBE_PROVIDERS // remove in production
+      BROWSER_APP_PROVIDERS // remove in production
     ])
       .catch(err => console.error(err));
   }

--- a/src/bootstrap.ts
+++ b/src/bootstrap.ts
@@ -1,7 +1,6 @@
 import { ComponentRef, PLATFORM_DIRECTIVES } from '@angular/core';
 import { bootstrap } from '@angular/platform-browser-dynamic';
 import { LocationStrategy, HashLocationStrategy } from '@angular/common';
-import { BROWSER_APP_PROVIDERS } from '@angular/platform-browser';
 import { ROUTER_DIRECTIVES } from '@angular/router';
 
 import { MODAL_BROWSER_PROVIDERS } from './components/angular2-modal/platform-browser';
@@ -23,7 +22,6 @@ export function main(): Promise<ComponentRef<App>> {
       {provide: PLATFORM_DIRECTIVES, multi: true, useValue: ROUTER_DIRECTIVES},
       APP_ROUTER_PROVIDERS,
       {provide: LocationStrategy, useClass: HashLocationStrategy},
-      BROWSER_APP_PROVIDERS // remove in production
     ])
       .catch(err => console.error(err));
   }

--- a/src/components/angular2-modal/package.json
+++ b/src/components/angular2-modal/package.json
@@ -22,10 +22,10 @@
     "url": "https://github.com/shlomiassaf/angular2-modal/issues"
   },
   "peerDependencies": {
-    "@angular/common": "^2.0.0-rc.2",
-    "@angular/compiler": "^2.0.0-rc.2",
-    "@angular/core": "^2.0.0-rc.2",
-    "@angular/platform-browser": "^2.0.0-rc.2",
-    "@angular/platform-browser-dynamic": "^2.0.0-rc.2"
+    "@angular/common": "^2.0.0-rc.4",
+    "@angular/compiler": "^2.0.0-rc.4",
+    "@angular/core": "^2.0.0-rc.4",
+    "@angular/platform-browser": "^2.0.0-rc.4",
+    "@angular/platform-browser-dynamic": "^2.0.0-rc.4"
   }
 }

--- a/src/components/angular2-modal/platform-browser/outside-event-plugin.ts
+++ b/src/components/angular2-modal/platform-browser/outside-event-plugin.ts
@@ -2,7 +2,7 @@
 // http://www.bennadel.com/blog/3025-creating-custom-dom-and-host-event-bindings-in-angular-2-beta-6.htm
 
 import {Injectable} from '@angular/core';
-import {DomEventsPlugin} from '@angular/platform-browser';
+import {DomEventsPlugin} from '@angular/platform-browser/src/dom/events/dom_events';
 import {getDOM, DomAdapter} from '@angular/platform-browser/src/dom/dom_adapter';
 import {noop} from '@angular/core/src/facade/lang';
 

